lib/reek/report/report.rb

```diff
@@ -63,14 +67,19 @@ def smells
         examiners.map(&:smells).flatten
       end
 
+      def progress(msg)
```

Comment:
I put this here as a nod towards the idea that other Reporters could easily add progress reporting. Happy to move it if you disagree.